### PR TITLE
Don't include files with source extension in None item by default

### LINKED ISSUE #1157: Setting EnableDefaultCompileItems to False doesn't remove the files from solution Explorer
_From @jinujoseph on February 8, 2017 23:9_

Create a ClassLibrary (.Net Core) 
Take a copy of Class1.cs and drop in the same folder
you will see of them included 
Now edit the csproj and put the below information 

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <EnableDefaultCompileItems>False</EnableDefaultCompileItems>
    <TargetFramework>netcoreapp1.0</TargetFramework>
  </PropertyGroup>
 <ItemGroup>
    <Compile Include="Class1.cs" />
 </ItemGroup>
</Project>
```
Build again , it includes only Class1.cs but the solution explorer shows all files, expected ? 

image

**Build**
26206.0.d15rel

_Copied from original issue: dotnet/project-system#1504_

### PR DESCRIPTION
**Pull request by dsplaisted**
Fixes #1157

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props [hunk 1]
```diff
@@ -28,7 +28,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   </ItemGroup>
   <ItemGroup Condition=" '$(EnableDefaultItems)' == 'true' And '$(EnableDefaultNoneItems)' == 'true' ">
     <None Include="**/*" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
-    <None Remove="**/*$(DefaultLanguageSourceExtension)" Condition=" '$(EnableDefaultCompileItems)' == 'true' "/>
+    <None Remove="**/*$(DefaultLanguageSourceExtension)" />
     <None Remove="**/*.resx" Condition=" '$(EnableDefaultEmbeddedResourceItems)' == 'true' "/>
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props]: Should we also be excluding resx from None regardless of EnableDefaultEmbeddedResourceItems. Seems like that would be more consistent.

Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props]: Good catch, I think we should also exclude .resx.

**Comment by dsplaisted** [comment]
@dotnet-bot test this please

File: src/Tests/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs [hunk 2]
```diff
@@ -366,6 +366,51 @@ public void It_allows_a_CSharp_file_to_be_used_as_Content()
...
+        {
+            var testProject = new TestProject()
+            {
+                Name = "DontIncludeSourceFilesInNone",
+                TargetFrameworks = "netcoreapp2.0",
+                IsExe = true,
+                IsSdkProject = true
+            };
+            testProject.AdditionalProperties["EnableDefaultCompileItems"] = "false";
+
```
Comment (nguerrera) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs]: Maybe add coverage for resource items too.

**Review by nguerrera — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props [hunk 3]
```diff
@@ -28,8 +28,8 @@ Copyright (c) .NET Foundation. All rights reserved.
   </ItemGroup>
   <ItemGroup Condition=" '$(EnableDefaultItems)' == 'true' And '$(EnableDefaultNoneItems)' == 'true' ">
     <None Include="**/*" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
-    <None Remove="**/*$(DefaultLanguageSourceExtension)" Condition=" '$(EnableDefaultCompileItems)' == 'true' "/>
-    <None Remove="**/*.resx" Condition=" '$(EnableDefaultEmbeddedResourceItems)' == 'true' "/>
+    <None Remove="**/*$(DefaultLanguageSourceExtension)" />
+    <None Remove="**/*.resx" />
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props]: Q: Does this traverse the project tree three times. I realize we're already doing it in common case. cc @cdmihai

**Comment by dsplaisted** [comment]
@dotnet-bot test Ubuntu14.04 Debug

**Comment by dsplaisted** [comment]
@dotnet-bot test Ubuntu14.04 Debug

**Comment by dsplaisted** [comment]
@MattGertz FYI for tell mode

**Pull request merged**